meilisearch-release-changelog/v1.4.0.md

```diff
@@ -0,0 +1,129 @@
+  -H 'Content-Type: application/json'  \
+  --data-binary '["@", "#", "&"]'
+```
+
+Done by @ManyTheFish in #3946
+
+## Load user-defined dictionaries
+
+Meilisearch word segmentation also relies on language-based dictionaries to segment words. Use the `dictionary` index setting to expand the default dictionaries and improve accuracy when working with datasets using domain-specific terms:
+
+```bash
+curl \
+  -X PUT 'http://localhost:7700/indexes/articles/settings/dictionary' \
+  -H 'Content-Type: application/json'  \
+  --data-binary '["J. R. R.", "J.R.R."]'
+```
+
+This feature can be used together with the `stopWords` and `synonyms` index settings:
+
+```json
+{
+    "dictionary": ["J. R. R.", "J.R.R."],
+    "synonyms": {
+            "J.R.R.": ["jrr", "J. R. R."],
+            "J. R. R.": ["jrr", "J.R.R."],
+            "jrr": ["J.R.R.", "J. R. R."],
+    }
+}
+```
+
+Done by @ManyTheFish in #3946
+
+## Other improvements
+
+Meilisearch better hides your data and no longer shows hidden document's fields in error messages. In case you are trying to sort by using a field that is not sortable but there are some of them that are not displayed either, you will see a message like the following:
+
+```
+Available sortable attributes are: price, stock, <..hidden-attributes>.
+```
+
+Done by @vivek-26 in #3945
+
+# Fixes 🐞
+
+* Fix synonyms display (#3946) @ManyTheFish
+* Fix thai synonyms (#4033) @Kerollmops
+* Fix synonyms with separators (#3994) @ManyTheFish
+* Fix the swap index tasks (#4041) @irevoire
+* Fix highlighting bug when searching for a phrase with cropping (#4028) @vivek-26
+* Empty arrays/objects now return empty instead of null (#3997) @dogukanakkaya
+* Fix the stats of the documents deletion by filter (#4053) @irevoire
+* `\` can now be used at the end of the `filter` search parameter values (#4038 and #4043) @Kerollmops
+
+## 🔴 Usage breaking change following `filter` bug fix
+
+⚠️ This section only concerns users using the `filter` search parameter with `\` characters in the filter expression.
+
+In this version, we fixed a bug regarding `\` in the `filter` search parameter expression. **Before v1.4.0, the users were not able to express all the filter expression they wanted.**
+
+For example, if you add the following documents:
+
+```json
+[
+  {
+    "id": 1,
+    "path": "my\\test\\path"
+  },
+  {
+    "id": 2,
+    "path": "my\\test\\path\\"
+  }
+]
+```
+
+(`\\` escaping here is about JSON escaping)
+
+Before v1.4.0, if you wanted to filter on document 2, the filters `path = "my\\test\\path\\"` and `path = "my\\test\\path\\\"` led to an error.
+
+Now, in v1.4.0, you can express all the possible filter expressions, no matter if the filter contains `\` or not. But to reach this, **you have to escape all the `\` characters in your filter**.
+
+Following our example, to finally succeed in filtering on document 2, the filter should be: `path = "my\\\\test\\\\path\\\\"`.
```

Comment:
On this line, the JSON and filtering escaping are visible:
`\\\\` is giving the filter a `\\` and the filter gives us a single `\`.